connectionpool.go

```diff
@@ -274,6 +278,11 @@ func (c *SimplePool) removeConnLocked(conn *Conn) {
 		delete(c.connPool, conn.addr)
 	}
 	delete(c.conns, conn)
+
+	// im not so keen on executing this with c.mu held
```

Comment:
I can see why you're not keen on this firing whilst the lock is being held - what are the pros and contras of this approach?